app/bundles/PageBundle/Controller/PageController.php

```diff
@@ -563,7 +569,12 @@ public function editAction(
                 $model->unlockEntity($entity);
             }
 
-            if ($cancelled || ($valid && $this->getFormButton($form, ['buttons', 'save'])->isClicked())) {
+            if ($cancelled
+                || ($valid && ($this->getFormButton($form, ['buttons', 'save'])->isClicked()
+                    || ($form->get('buttons')->has('save_draft') && $this->getFormButton($form, ['buttons', 'save_draft'])->isClicked())
+                    || ($form->get('buttons')->has('apply_draft') && $this->getFormButton($form, ['buttons', 'apply_draft'])->isClicked())
+                    || ($form->get('buttons')->has('discard_draft') && $this->getFormButton($form, ['buttons', 'discard_draft'])->isClicked())))
+            ) {
```

Comment:
```suggestion
            if ($cancelled
                || ($valid && $this->isAnyOfButtonsClicked($form, ['save', 'save_draft', 'apply_draft', 'discard_draft']))
            ) {
```
Let's reduce the duplication and improve readability by extracting the logic into methods as follows.
```php
private function isButtonClicked(FormInterface $form, string $name): bool
{
    return $form->get('buttons')->has($name) && $this->getFormButton($form, ['buttons', $name])->isClicked();
}

/**
 * @param string[] $names
 */
private function isAnyOfButtonsClicked(FormInterface $form, array $names): bool
{
    foreach ($names as $name) {
        if ($this->isButtonClicked($form, $name)) {
            return true;
        }
    }

    return false;
}
```